src/pages/ForgotPassword/forgotpassword.spec.js

```diff
@@ -0,0 +1,185 @@
+};
+
+const renderWithEnzymes = state => {
+  store = mockStore({
+    authReducer: {
+      error: null,
+      isAuthenticated: false,
+      user: {},
+      status: 'rest'
+    },
+    resetPassword: {
+      ...state
+    }
+  });
+  return mount(
+    <Provider store={store}>
+      <BrowserRouter>
+        <Switch>
+          <RequestPasswordResetPage {...props} />
+        </Switch>
+      </BrowserRouter>
+    </Provider>
+  );
+};
+
+const renderWithRTL = state => {
+  store = mockStore({
+    authReducer: {
+      error: null,
+      isAuthenticated: false,
+      user: {},
+      status: 'rest'
+    },
+    resetPassword: {
+      ...state
+    }
+  });
+  return {
+    ...render(
+      <Provider store={store}>
+        <BrowserRouter>
+          <Switch>
+            <RequestPasswordResetPage {...props} />
+          </Switch>
+        </BrowserRouter>
+      </Provider>
+    ),
+    store
+  };
+};
+
+describe('Test forgot password page', () => {
+  afterEach(cleanup);
+
+  it('should have a email address field', () => {
+    const wrapper = renderWithEnzymes();
+    const field = wrapper.find('input').first();
+    expect(field.exists()).toBe(true);
+    expect(field.props().name).toEqual('email');
+  });
+
+  it('should have a submit button', () => {
+    const wrapper = renderWithEnzymes();
+    const button = wrapper.find('button').first();
+    expect(button.exists()).toBe(true);
+  });
+
+  it('should render forgot password page', () => {
+    const wrapper = renderWithEnzymes();
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+
+  it('should update password field on change', () => {
+    const wrapper = renderWithEnzymes();
+    const passwordInput = wrapper.find("input[name='email']");
+    passwordInput.simulate('change', {
+      persist: () => {},
+      target: {
+        name: 'email',
+        value: 'tejumoladavid@gmail.com'
+      }
+    });
+    expect(passwordInput.html()).toMatch('tejumoladavid@gmail.com');
+  });
+
+  test('should submit a valid form', async () => {
+    const { getByTestId, findByText } = renderWithRTL(initialSate);
+
+    const emailInputElement = getByTestId('email');
+    fireEvent.change(emailInputElement, {
+      target: { value: 'tejumoladavid@gmail.com' }
+    });
+    fireEvent.blur(emailInputElement);
+
+    const formNode = getByTestId('submit-form');
+    fireEvent.submit(formNode);
+
+    await waitForDomChange(() => {
+      const submitting = findByText('SUBMITTING');
+      expect(submitting).toBeTruthy();
+      expect(props.handleSubmit).toHaveBeenCalled();
+    });
+  });
+
+  it('should return error if email is invalid', () => {
+    expect(validateEmail('tejumoladavid')).toEqual('Invalid email address');
+  });
+
+  it('should return error if email is null', () => {
+    expect(validateEmail(null)).toEqual('Email address is required');
+  });
+
+  it('should render loader when submit is clicked', () => {
+    const wrapper = renderWithEnzymes();
```

Comment:
from what you stated in the "it" block, there is no assertion that the submit button is clicked to render the loader.